src/schema/sources.ts

```diff
@@ -1483,6 +1491,20 @@ export const resolvers: IResolvers<any, Context> = {
         inputHandle !== current.handle,
       );
 
+      if (!isNullOrUndefined(isPrivate) && current.private !== isPrivate) {
+        const approved = await ctx.con
+          .getRepository(SquadPublicRequest)
+          .findOne({
+            where: { sourceId, status: SquadPublicRequestStatus.Approved },
+          });
+
+        if (!approved) {
+          throw new ValidationError(
+            'Squad has not been approved yet of becoming public',
```

Comment:
Add the error message to `errors.ts` file

Reply:
Oh cool. Let me move this. Thank you!